bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -220,25 +248,121 @@ pub mod pallet {
 			Self::deposit_event(Event::ExportOperatingModeChanged { mode });
 			Ok(())
 		}
+
+		/// Add an additional relayer tip for a committed message identified by `message_id`.
+		/// The tip asset will be swapped for ether.
+		#[pallet::call_index(2)]
+		#[pallet::weight(
+			T::WeightInfo::add_tip()
+				.saturating_add(T::BackendWeightInfo::transact_add_tip())
+		)]
+		pub fn add_tip(
+			origin: OriginFor<T>,
+			message_id: MessageId,
+			asset: Asset,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let ether_location = T::EthereumLocation::get();
+			let (tip_asset_location, tip_amount) = match asset {
+				Asset { id: AssetId(ref loc), fun: Fungibility::Fungible(amount) } => (loc, amount),
+				_ => return Err(Error::<T>::UnsupportedAsset.into()),
+			};
+
+			let ether_gained = if *tip_asset_location != ether_location {
+				Self::swap_and_burn(
+					who.clone(),
+					tip_asset_location.clone(),
+					ether_location,
+					tip_amount,
+				)
+				.map_err(|_| Error::<T>::SwapError)?
+			} else {
+				tip_amount
+			};
+
+			// Send the tip details to BH to be allocated to the reward in the Inbound/Outbound
+			// pallet
+			let dest = T::BridgeHubLocation::get();
+			let call = Self::build_add_tip_call(who.clone(), message_id.clone(), ether_gained);
+			let remote_xcm = Self::build_remote_xcm(&call);
+			let local_pallet_origin: Location = T::PalletLocation::get().into();
+			let who_location =
+				T::AccountToLocation::try_convert(&who).map_err(|_| Error::<T>::InvalidAccount)?;
+
+			let xcm_message_id =
+				Self::send_xcm(local_pallet_origin, who_location, dest.clone(), remote_xcm.clone())
```

Comment:
See comment below regarding `send_xcm`.

`register_token` and `add_tip` have the same behavior when it comes to charging xcm delivery fees - from the user who called the extrinsic.



```suggestion
				Self::send_xcm(who_location, dest.clone(), remote_xcm.clone())
```

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/23/commits/75fb25b52d6d19c910d48921db556e1cd2e5ab40